### PR TITLE
Refactoring of vertex model

### PR DESCRIPTION
**Pull request by 3nids**

### CODE REVIEW

**Comment by qfield-fairy** [comment]
Uploaded test apks for
 - [**armv7**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-1040-5d1bbb1929619c5d391a9d8588391b8e58bae0b2-armv7.apk)
 - [**x86**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-1040-5d1bbb1929619c5d391a9d8588391b8e58bae0b2-x86.apk)
 - [**arm64_v8a**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-1040-5d1bbb1929619c5d391a9d8588391b8e58bae0b2-arm64_v8a.apk)
 - [**x86_64**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-1040-5d1bbb1929619c5d391a9d8588391b8e58bae0b2-x86_64.apk)

**Comment by qfield-fairy** [comment]
Uploaded test apks for
 - [**armv7**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-1040-c5cee877760b3ad0ccc36aa3ae38ef011db8f7a9-armv7.apk)
 - [**x86**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-1040-c5cee877760b3ad0ccc36aa3ae38ef011db8f7a9-x86.apk)
 - [**arm64_v8a**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-1040-c5cee877760b3ad0ccc36aa3ae38ef011db8f7a9-arm64_v8a.apk)
 - [**x86_64**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-1040-c5cee877760b3ad0ccc36aa3ae38ef011db8f7a9-x86_64.apk)

**Comment by qfield-fairy** [comment]
Uploaded test apks for
 - [**armv7**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-1040-2dc8e39ab7c2e8451dcd513f8279ae86cec4cd67-armv7.apk)
 - [**x86**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-1040-2dc8e39ab7c2e8451dcd513f8279ae86cec4cd67-x86.apk)
 - [**arm64_v8a**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-1040-2dc8e39ab7c2e8451dcd513f8279ae86cec4cd67-arm64_v8a.apk)
 - [**x86_64**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-1040-2dc8e39ab7c2e8451dcd513f8279ae86cec4cd67-x86_64.apk)

File: src/core/rubberbandmodel.cpp [hunk 1]
```diff
@@ -187,7 +187,7 @@ QgsPoint RubberbandModel::currentCoordinate() const
 void RubberbandModel::setCurrentCoordinate( const QgsPoint &currentCoordinate )
 {
   // play safe, but try to find out
-  Q_ASSERT(mPointList.count() != 0);
+  // Q_ASSERT( mPointList.count() != 0 );
```
[the remark this answers is not in the document]
Comment (3nids) [reply src/core/rubberbandmodel.cpp:190]: not sure, I could not track this down.

**Comment by qfield-fairy** [comment]
Uploaded test apks for
 - [**armv7**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-1040-34f7468fa04690962cd3773909e265cd4f4f47f6-armv7.apk)
 - [**x86**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-1040-34f7468fa04690962cd3773909e265cd4f4f47f6-x86.apk)
 - [**arm64_v8a**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-1040-34f7468fa04690962cd3773909e265cd4f4f47f6-arm64_v8a.apk)
 - [**x86_64**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-1040-34f7468fa04690962cd3773909e265cd4f4f47f6-x86_64.apk)

**Pull request merged**